staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -162,7 +162,11 @@ func GetPrefixTransformers(config *ResourceConfig) ([]value.PrefixTransformer, e
 			}
 
 			// Get gRPC client service with endpoint.
-			envelopeService, err := envelopeServiceFactory(provider.KMS.Endpoint, kmsPluginConnectionTimeout)
+			timeout := kmsPluginConnectionTimeout
+			if provider.KMS.Timeout.Duration != 0 {
```

Comment:
`if provider.KMS.Timeout != nil && provider.KMS.Timeout.Duration != 0 {`

Reply:
check `provider.KMS.Timeout`, not `provider.KMS`

Reply:
I believe that would result in compile error:
cannot convert nil to type v1.Duration

I think that the original version was correct since the zero value for Timeout.Duration is 0.
The second test in the unit test simulates the scenario were timeout is not supplied.